pisces/data_sets.py

```diff
@@ -294,26 +322,24 @@ def load_feature_data(self, feature: str | None, id: str | None) -> Dict[str, np
     
     @classmethod
     def find_data_sets(cls, root: str | Path) -> Dict[str, 'DataSetObject']:
-        set_dir_regex = r".*" + cls.FEATURE_PREFIX + r"(.+)"
-        # this regex matches the feature directory name and the data set name
-        # but doesn't work on Windows (? maybe, cant test) because of the forward slashes
-        feature_dir_regex = r".*/(.+)/" + cls.FEATURE_PREFIX + r"(.+)"
+        root = str(root).replace("\\", "/") # Use consistent separators
+        if not root.endswith("/"):
+            root += "/"
+
+        feature_dir_regex = rf".*/(.+)/{cls.FEATURE_PREFIX}(.+)/?"
```

Comment:
I actually think the `/?` on the end makes `if not root.endswith("/"): root += "/"` redundant. Let's discuss!